ddtrace/appsec/processor.py

```diff
@@ -204,9 +207,45 @@ def __attrs_post_init__(self):
         # we always need the response headers
         self._mark_needed(_Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES)
 
-    def on_span_start(self, span):
-        # type: (Span) -> None
-        pass
+    def update_rules(self, new_rules):
+        # type: (List[Dict[str, Any]]) -> None
+        self._ddwaf.update_rules(new_rules)
+
+    def on_span_start(self, span, *args, **kwargs):
+        # type: (Span, Any, Any) -> None
+        peer_ip = kwargs.get("peer_ip")
```

Comment:
what are the other options you can think of for us to get this information here?

I don't really like that we are extending the public API for creating traces to meet an internal only need.


I am assuming we are not using span tags because this is information we don't want to collect in tags? (would always have to be removed before encoding the span whether we had ASM enabled or not?)

Reply:
PR updated, I switched to pass the early data using contextvars until the new context API allows for passing data without a span, please take a look.

Reply:
@brettlangdon could you remove the required change from here if you are ok with the new context manager for contextvars?